### ISSUE TITLE
[FEATURE REQUEST] Export Page Order

### ISSUE THREAD

**Issue by theatischbein** [Akylas/OSS-DocumentScanner#381]
**Is your feature request related to a problem? Please describe.**
If I want to export or print multiple scanned documents together there is no way to set the order they appear in the document. The latest document will always be the first. 

**Describe the solution you'd like**
There should be an export order selectable in which the documents appear in the export.

**Describe alternatives you've considered**
There is the option to enter page number in the printing dialog and the option to 'Print to PDF', but even if I enter 4,3,2,1 there the document is exported in the wrong order.

**Additional context**
Even renaming the documents does not change the showed and selectable order. Maybe the problem could also be by allowing to reorder scanned documents ?

**Comment by farfromrefug** [comment]
@theatischbein thanks for reporting the issue i will look at it

**Comment by farfromrefug** [comment]
@theatischbein i have looked into this:
* i cant reproduce the export having latest document first. Could you share a video to show the error? Also please share your version numbers
* about the printing dialog, it is actually a system printing dialog. Indeed it does not work when you try to change the pages order. Cant really do anything about this. I have been thinking about this and i dont think i will add a way to change order of pages (except fixing the first point once i can reproduce it) because there are too many special cases which would make it not work too many times. Hope you can understand

**Comment by theatischbein** [comment]
First I select them in the main feed and export them as PDF.

https://github.com/user-attachments/assets/759631c8-c533-4269-8e02-fe6f8eb28b55

The result is this PDF, where the upper image is the first page.
[1736259132602.pdf](https://github.com/user-attachments/files/18333758/1736259132602.pdf)


Than I moved both into the same folder.
I renamed than, so that 001 should be before 002.
Now I select them in the other way.

https://github.com/user-attachments/assets/89676985-dbee-49de-8a5a-3d016cd0887b

The result is this PDF, but 002 is still the first page.
[1736259196001.pdf](https://github.com/user-attachments/files/18333780/1736259196001.pdf)

**Comment by theatischbein** [comment]
My use case is the following:

- I want to scan a multi page document
- I usually start by the first page and do the others afterwards
- If I want to export them now the first page will always be the last in the PDF
- I would need to start scanning the last page first and go through by document in reverse order

Is there an easier way to scan and export a multi page document ?

Ordering the photos by name for export would be one solution to this problem.

I don't know if Android supports to track which photo I select first when multi selecting and exporting

**Comment by farfromrefug** [comment]
@theatischbein First you can add pages to an existing document (not sure if it what you are doing, or if you are creating a new "document" each time):
* start scanning with "+" from the main view => new document
* go into a document from the main view and use "+"=> added to that document.

The second case  is what you should do. Pages will be ordered by "added date" So if you do page 1,2,3.... it should always export in the right order.
I dont really see in which case you would end up with last page first. Remember page order = scan order.

I will think about adding sort option for export.

**Comment by theatischbein** [comment]
> Remember page order = scan order.

Yes that is exactly point of this issue. I get why it is implemented that way, but I find it very unintuitive. 

Maybe it would be helpful to add an Option like "reverse page order" for the export settings ?

**Comment by farfromrefug** [comment]
@theatischbein i will add a "sort option". Will be available for "viewing"(main view) and "exporting to pdf". Might just take a bit of time as not really sure how to integrate it UX wise. The app now has a lot of buttons/options. Must keep it clean

**Comment by maninthemiddle01** [comment]
Maybe in the three dots menu with a symbol or text like "sort by". And maybe there could also be the possibility to reorder individual pages on long press and drag and drop?

**Comment by farfromrefug** [comment]
@maninthemiddle01 in the next version exporting multiple documents will always respect the current sort order
